src/workerd/io/actor-sqlite.h

```diff
@@ -156,13 +161,57 @@ class ActorSqlite final: public ActorCacheInterface, private kj::TaskSet::ErrorH
   // transactions should be used in the meantime.
   kj::OneOf<NoTxn, ImplicitTxn*, ExplicitTxn*> currentTxn = NoTxn();
 
+  // Backs the `kj::Own<void>` returned by `armAlarmHandler()`.
+  class DeferredAlarmDeleter: public kj::Disposer {
+  public:
+    // The `Own<void>` returned by `armAlarmHandler()` is actually set up to point to the
+    // `ActorSqlite` itself, but with an alternate disposer that deletes the alarm rather than
+    // the whole object.
+    void disposeImpl(void* pointer) const {
+      reinterpret_cast<ActorSqlite*>(pointer)->maybeDeleteDeferredAlarm();
+    }
+  };
+
+  struct UnknownAlarmTime {};
+  struct KnownAlarmTime {
+    enum class Status {
+      CLEAN,    // Alarm time has been committed
+      DIRTY,    // Alarm time has been changed and not yet committed
+      FLUSHING  // Alarm time is being committed in an ongoing transaction
+    } status;
+    kj::Maybe<kj::Date> time;
+  };
+  struct DeferredAlarmDelete {
```

Comment:
Could use some more commentary here on what this is for. It took me a while to realize this is when we're currently running the alarm handler, and we need to delete the alarm at the end of the handler unless it is changed during the handler...